### DISCUSSION TITLE
[BUG] `RuntimeError: unauthorized use of script`

### DISCUSSION THREAD

**Discussion by Itakkkkkk** [dashingsoft/pyarmor#1259] [Q&A]
Version: 8.2
I tried to obfuscate python scripts in subdirectories with the following script.
```
cnt=0
for dir in $(find $1 -mindepth 1 -maxdepth 1 -type d)
do 
    poetry run pyarmor cfg restrict_module=0
    if [ $cnt -eq 0 ]; then
        poetry run pyarmor gen --platform linux.aarch64 -r $dir -O dist
    else
        poetry run pyarmor gen --no-runtime --outer --platform linux.aarch64 -r $dir -O dist
    fi
    cnt=1
done
```
Obfuscating is completed successfully but I got `RuntimeError: unauthorized use of script`at runtime.

**Comment by jondy** [comment]
Please use pyarmor by this way
```
poetry run pyarmor gen --platform linux.aarch64 -r -O dist $dir/*.py $dir/dir1 $dir/dir2 ...
```

**Reply by tarikwaleed** [reply]
do i have to specify each python file and subpackage separately? 
isn't the -r flag sufficient for recursive obfuscation?

**Reply by Warfront1** [reply]
I was never able to get the `-r` option to work as I would have expected, but I have found a workaround.
When I use the pyarmor gen command I typically like to place my python scripts into a folder.
`python -m pyarmor.cli gen -O obfdist src/`
In the above command I am generating obfuscated scripts for all python scripts found in `src` including scripts inside of deeper folder structures.
I would recommend putting test code to ensure obfuscating in various Python scripts throughout your folder structure to ensure complete obfuscation.

**Reply by tarikwaleed** [reply]
yes, that's what i've done. i put all of my code under src/mypkg then run the command `python gen -r src/mypkg` and the obfuscation completed successfully.
also make sure to set the option `restrict_module` to `0` by running the command ` pyarmor cfg restrict_module=0`
